enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/grpc/ExtAuthService.java

```diff
@@ -164,16 +164,18 @@ private CheckResponse buildResponse(CheckRequest request, ResponseObject respons
                 HeaderValueOption headerValueOption = HeaderValueOption.newBuilder()
                         .setHeader(HeaderValue.newBuilder().setKey(APIConstants.PATH_HEADER).setValue(constructedPath)
                                 .build()).build();
-                okResponseBuilder.addHeaders(headerValueOption);
+                    okResponseBuilder.addHeaders(headerValueOption);
             }
 
             if (responseObject.getHeaderMap() != null) {
                 responseObject.getHeaderMap().forEach((key, value) -> {
                             HeaderValueOption headerValueOption = HeaderValueOption.newBuilder()
                                     .setHeader(HeaderValue.newBuilder().setKey(key).setValue(value).build())
                                     .build();
-                            okResponseBuilder.addHeaders(headerValueOption);
-                        }
+                            if (!(headerValueOption.getHeader().getValue().equals("wso2_lambda_egress_gateway"))) {
+                                okResponseBuilder.addHeaders(headerValueOption);
+                            }
```

Comment:
I think it is safe to check the key too. Let's check the key equals cluster header and values equals wso2_lambda_egress_gateway. We can use (key, value) directly in the forEach loop.

Reply:
Fixed. [https://github.com/wso2/product-microgateway/pull/3278/commits/7530563cefd13bea91da379804770bc0a9c8f99f](https://github.com/wso2/product-microgateway/pull/3278/commits/7530563cefd13bea91da379804770bc0a9c8f99f)